configure.d/1_vfs_ioctl.conf

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+. $(dirname $3)/conf_framework
+
+check() {
+    cur_name=$(basename $2)
+    config_file_path=$1
+    if compile_module $cur_name "vfs_ioctl(NULL, 0, 0);" "linux/fs.h"
+    then
+        echo $cur_name "1" >> $config_file_path
+    else
+        echo $cur_name "2" >> $config_file_path
+    fi
+}
+
+apply() {
+    case "$1" in
+    "1")
+        add_function "
+        static inline int cas_vfs_ioctl(struct file *file, unsigned int cmd,
+                unsigned long arg)
+        {
+            return vfs_ioctl(file, cmd, arg);
+        }" ;;
+    "2")
+        add_function "
+        static inline int cas_vfs_ioctl(struct file *file, unsigned int cmd,
+                unsigned long arg)
+        {
+            int error = -ENOTTY;
+
+            if (!file->f_op->unlocked_ioctl)
+                goto out;
+
+            error = file->f_op->unlocked_ioctl(file, cmd, arg);
+            if (error == -ENOIOCTLCMD)
+                error = -ENOTTY;
```

Comment:
why ENOTTY?

Reply:
It's returned by `vfs_ioctl()` on kernels where this function is available, so I made it consistent.